src/services/pool/pool.repository.ts

```diff
@@ -31,7 +31,15 @@ export default class PoolRepository {
   private initializeDecoratedAPIRepository() {
     return {
       fetch: async (): Promise<Pool[]> => {
-        return balancerAPIService.pool.get(this.queryArgs);
+        const pools = await balancerAPIService.pool.get(this.queryArgs);
+
+        const poolDecorator = new PoolDecorator(pools);
+        const decoratedPools = await poolDecorator.decorate(
+          this.tokens.value,
+          false
+        );
```

Comment:
Is this still needed in the repository or can it be restricted only to usePoolQuery?

Reply:
Good point will remove.